### ISSUE TITLE
Accessibility violation for ibm-table when there is no table data

### ISSUE THREAD

**Issue by GSSCDylan** [IBM/carbon-components-angular#1586]
## Detailed description

When using ibm-table, and there is no data for the table, IBM Equal Access Accessibility Checker says that "Complex table does not have headers for each cell properly defined with 'header' or 'scope'"

> Describe in detail the issue you're having.

> Is this issue related to a specific component?

- **ibm-table**

> What did you expect to happen? What happened instead? What would you like to see changed?

- **when there is no data, there should be headers for the  table**

> What browser are you working in?

- **FireFox**

> What version of the Carbon Design System are you using?

`"carbon-components": "10.20.0", "carbon-components-angular": "4.17.0"`

## Steps to reproduce the issue
1. Use firefox to open a url that contains ibm-table
    such as https://angular.carbondesignsystem.com/?path=/story/components-table--with-no-data
2. Make sure there is no table data
3. Run IBM Equal Access Accessibility Checker to scan AVT1 issues

## Additional information

* Screenshots or code

<img width="1292" alt="image" src="https://user-images.githubusercontent.com/73825731/97862232-f1189f80-1d3f-11eb-9398-3422254be02a.png">

![Uploading image.png…]()

**Comment by GSSCDylan** [comment]
Hi @cal-smith , please also help to fix this issue ASAP, it's also impact our AD AVT compliance. @stanislavgeorgiev  FYI

**Comment by stanislavgeorgiev** [comment]
It looks like a duplicate of this defect has already been merged > #1617
It was confirmed by out AVT team

**Comment by GSSCDylan** [comment]
Fixed in carbon-components-angular@4.31.1, thanks!
